Feature/controller specs

RileyManda opened the pull request:
I have completed implementing controller specs according to stipulated [requirements](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/controller_specs_project.md): 🌱 

# Implementations:

### Specs
- Added RSpec as a gem to your project.
- Created a Request spec file for all your controllers.

### Checks
- If response status was correct.
- If a correct template was rendered.
- If the response body includes correct placeholder text.


 In addition I have ensured the following general requirements:
 
### General requirements

> Made sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
> Made sure that I have used correct flow [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
> Made sure that I have documented my work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).
> Followed list of [best practices for Ruby](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md).


Please kindly review my implementation.

Thank you 🙏🏻

RileyManda committed:
Installed Rspec rails

RileyManda committed:
Implemented Posts|Users specs

RileyManda committed:
Implemented initial path request spec for users|posts

RileyManda committed:
Post requests specs

RileyManda committed:
Posts request specs

thecodechaser commented on spec/requests/posts_spec.rb:
```diff
@@ -0,0 +1,21 @@
+
+RSpec.describe 'Posts', type: :request do
+  let(:first_user) { User.create(name: 'John Doe', bio: 'Sample bio') }
+
+  describe 'GET /posts' do
+    it 'returns a 200 OK response' do
+      get posts_path
+      expect(response).to have_http_status(200)
+    end
+
+    it "contains 'Posts List' in the response body" do
+      get posts_path
+      expect(response.body).to include('Posts List')
+    end
+    it "renders the 'posts/index' template" do
+      get posts_path
+      expect(response).to render_template('posts/index')
+    end
```
- [x] Please make sure to add the three tests for the `posts#show` action as well as you added for the `posts#index` action as requested in the project requirements.

thecodechaser reviewed (changes requested):
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @RileyManda👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- Good job with controller specs. ✔️
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- No linter errors. :heavy_check_mark:
- PR has a good title and summary. ✔️

<img src="https://media2.giphy.com/media/Jq824R93JsLwZCaiSL/giphy.gif"/>

## Required Changes ♻️

- [ ]  The issues page link in the README file is broken, Please make sure to fix this. It will make the README file more professional.

- [ ]  Check the comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

RileyManda committed:
Implemented 3 additional tests for show:posts spec

RileyManda commented:
I have completed the iterations to the projects '  Posts spec as requested.

# Iterations:

### Specs
- Added 3 additional tests for the posts#show 
               - If response status is correct.
               - If a correct template was rendered.
               - If the response body includes correct placeholder text.

### Documentation

- fixed broken issues page link in readme file
- Added project description:Readme file


In addition I have ensured the following general requirements:
 
### General requirements

> Made sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
> Made sure that I have used correct flow [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
> Made sure that I have documented my work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).
> Followed list of [best practices for Ruby](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md).


Please kindly review my implementation.

Thank you 🙏🏻

RileyManda committed:
Fixed broken issue page url in readme file

RileyManda committed:
Updated read description

Strangeal reviewed (approved):
Hi @RileyManda, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

![](https://media.giphy.com/media/v1.Y2lkPTc5MGI3NjExZjk5MjNhMmYyM2RjZmZmNWRkNjA3OWNmNGFhMjI4NDViOTg4N2UzNyZlcD12MV9pbnRlcm5hbF9naWZzX2dpZklkJmN0PXM/x3l1BWfB85i2uU8nKK/giphy.gif)

## Highlights
- Add `spec` for show and index  actions ✅ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
